### PR TITLE
data.md: Remove excess chops for monotonically decreasing word ops

### PR DESCRIPTION
**Pull request by MrChico**
If you agree with the reasoning of https://github.com/kframework/evm-semantics/issues/386

### CODE REVIEW

**Comment by ehildenb** [comment]
I think this is reasonable, will post some profiling numbers on here with these changes.

Can you add some explanation before the definition of the various `_*Word_` operators which make this assumption explicit? Something like "for all these operators, it's assumed that the operands are already in the quotient ring/bounded".

File: data.md [hunk 1]
```diff
@@ -234,7 +234,7 @@ You could alternatively calculate `I1 modInt I2`, then add one to the normal int
 ```
 
 The corresponding `<op>Word` operations automatically perform the correct modulus for EVM words.
-
+Operands are assumed to be within the range of a 256 bit EVM word.
```
Comment (ehildenb) [inline data.md]: Formatting (blank line before code blocks). Otherwise line-numbers don't line up between .md and .k files.

Comment (ehildenb) [reply data.md]: Also you have a whitespace error at the end of this line.

**Comment by ehildenb** [comment]
No appreciable performance differences.

```
* e0a2cd6d - data.md: Remove excess chops for monotonically decreasing word ops               - Martin Lundfall
| testset: profile - 1
| 0  0.10s    0.05s     0.05s    15948KB     make  clean
| 0  0.03s    0.00s     0.02s    3440KB      rm    -rf               /mnt/ramdisk/klab
| 0  128.02s  215.47s   12.23s   831500KB    make  llvm-deps         -B
| 0  54.32s   403.42s   16.77s   2225168KB   make  build-java        build-llvm         build-ocaml                  -B  -j6
| 0  0.13s    0.10s     0.03s    15884KB     make  split-tests       -B
| 0  278.79s  2273.65s  884.77s  4488544KB   make  test-conformance  -j16               TEST_CONCRETE_BACKEND=ocaml
| 0  142.29s  1682.44s  460.52s  15364928KB  make  test-conformance  -j16               TEST_CONCRETE_BACKEND=llvm
| 0  584.55s  9733.74s  547.26s  2906008KB   make  test-proof        -j10
| 0  517.43s  736.84s   16.30s   3720028KB   make  test-klab-prove   -j4
| testset: profile - 2
| 0  0.11s    0.06s     0.04s    16000KB     make  clean
| 0  0.03s    0.00s     0.02s    3528KB      rm    -rf               /mnt/ramdisk/klab
| 0  128.22s  215.09s   12.46s   820728KB    make  llvm-deps         -B
| 0  55.58s   421.18s   17.19s   2274248KB   make  build-java        build-llvm         build-ocaml                  -B  -j6
| 0  0.13s    0.10s     0.03s    16008KB     make  split-tests       -B
| 0  279.89s  2287.72s  890.72s  4488572KB   make  test-conformance  -j16               TEST_CONCRETE_BACKEND=ocaml
| 0  142.86s  1686.50s  463.87s  15365456KB  make  test-conformance  -j16               TEST_CONCRETE_BACKEND=llvm
| 0  567.96s  9588.13s  529.33s  3499972KB   make  test-proof        -j10
| 0  521.83s  738.10s   16.44s   3723412KB   make  test-klab-prove   -j4
| 
* 39c92c4e - deps/k: update submodule (#383)                                                  - rv-jenkins
| testset: profile - 1
| 0  0.09s    0.06s     0.02s    15944KB     make  clean
| 0  0.02s    0.00s     0.02s    3388KB      rm    -rf               /mnt/ramdisk/klab
| 0  128.05s  214.16s   12.29s   838528KB    make  llvm-deps         -B
| 0  54.33s   408.45s   15.97s   2191880KB   make  build-java        build-llvm         build-ocaml                  -B  -j6
| 0  0.13s    0.10s     0.03s    15952KB     make  split-tests       -B
| 0  276.84s  2270.87s  891.10s  4488536KB   make  test-conformance  -j16               TEST_CONCRETE_BACKEND=ocaml
| 0  142.40s  1679.35s  464.03s  15365336KB  make  test-conformance  -j16               TEST_CONCRETE_BACKEND=llvm
| 0  574.02s  9779.07s  528.77s  3276448KB   make  test-proof        -j10
| 0  523.43s  744.99s   18.44s   3732508KB   make  test-klab-prove   -j4
| testset: profile - 2
| 0  0.10s    0.06s     0.03s    16048KB     make  clean
| 0  0.02s    0.00s     0.02s    3520KB      rm    -rf               /mnt/ramdisk/klab
| 0  128.82s  216.44s   12.34s   819284KB    make  llvm-deps         -B
| 0  55.95s   417.98s   17.62s   2167300KB   make  build-java        build-llvm         build-ocaml                  -B  -j6
| 0  0.14s    0.10s     0.03s    16000KB     make  split-tests       -B
| 0  278.26s  2281.58s  895.00s  4488644KB   make  test-conformance  -j16               TEST_CONCRETE_BACKEND=ocaml
| 0  142.52s  1683.59s  462.41s  15364732KB  make  test-conformance  -j16               TEST_CONCRETE_BACKEND=llvm
| 0  554.32s  9445.97s  513.70s  3053388KB   make  test-proof        -j10
| 0  528.86s  755.55s   18.87s   3744800KB   make  test-klab-prove   -j4
| testset: profile - 3
| 0  0.11s    0.06s     0.05s    15860KB     make  clean
| 0  0.03s    0.00s     0.02s    3500KB      rm    -rf               /mnt/ramdisk/klab
| 0  128.08s  215.50s   12.27s   816492KB    make  llvm-deps         -B
| 0  55.53s   424.06s   16.56s   2175988KB   make  build-java        build-llvm         build-ocaml                  -B  -j6
| 0  0.12s    0.08s     0.04s    15952KB     make  split-tests       -B
| 0  279.06s  2287.44s  893.95s  4488608KB   make  test-conformance  -j16               TEST_CONCRETE_BACKEND=ocaml
| 0  143.13s  1690.58s  464.68s  15365020KB  make  test-conformance  -j16               TEST_CONCRETE_BACKEND=llvm
| 0  579.51s  9824.37s  536.06s  3060032KB   make  test-proof        -j10
| 0  527.46s  750.06s   18.68s   3720180KB   make  test-klab-prove   -j4

```

**Comment by MrChico** [comment]
Makes sense that there is no noticable change. There might be one if one uses the Integer operations instead of Word operations where it possible, such as for `AND`, `OR` or `XOR`. This would also be less LOC, less productions and rules. 
It would make arithmetic opcodes look a little less uniform though. Would you like me to try this?

My main purpose behind this was only to get rid of these ugly superfluous `chop`, not any performance concerns.

**Comment by ehildenb** [comment]
No, I'd rather go through the `*Word` operations, the idea being that we can change out the internal representation of `Word` just by changing out `data.md`, not by modifying the semantics.

I agree that removing the ugly superfluous chops is nice, and would like to merge this.

Can you address my review? You have a formatting error. Also, I'd ilke a little more detailed of an explanation, something along the lines of a _warning_ to the user "make sure you know that the operands are already in the quotient field".

**Comment by MrChico** [comment]
Addressed

**Comment by MrChico** [comment]
Ready to merge?

**Comment by ehildenb** [comment]
Well tests have to pass. But I'm re-running them.

**Comment by MrChico** [comment]
@ehildenb sweett

**Pull request merged**